app/models/translation.rb

```diff
@@ -4,4 +4,48 @@ class Translation < ActiveRecord::Base
 	attr_accessible :user, :video
 
 	validates_presence_of :video
+
+	def video
+		Video.find(self.video_id)
+	end
+
+	def self.create_and_save_new_translation(video, user, srt)
+		self.verify_file(srt)
+		root_folder_name = "public/srts/"
+		self.make_folder(root_folder_name)
+		video_folder_name = "#{root_folder_name}#{video.id}"
+		self.make_folder(video_folder_name)
+		self.remove_srt_if_exists(video, user, video_folder_name)
+		translation = Translation.create(:video => video, :user => user)
+		file_name =  "#{translation.id}.srt"
+    	path = File.join(video_folder_name, file_name)
+    	File.open(path, "wb") { |f| f.write(srt.read) }
+		return translation
+	end
+
+	def self.remove_srt_if_exists(video, user, video_folder_name)
+		other_records = Translation.where(:user_id => user.id, :video_id => video.id)
```

Comment:
Also note that we'll indirectly enforce that each user can only have one translation per video. That makes this `other_records` variable redundant.